docs/studios/unity/unity-sdk/RGAction.mdx

```diff
@@ -1,116 +1,172 @@
+{
+    // Your logic here
+}
+```
 
-#### Return Value
+</TabItem>
+</Tabs>
 
-None
+The `RGAction` attribute is designed for ease of integration.
+A method with this attribute may execute an action from start to finish, or it may start an action that `MonoBehavior` methods such as `FixedUpdate` should complete.
+Depending on how your code is structured, you may be able to add the `RGAction` attribute to your existing code with minimal refactoring.
 
-#### Examples
 <Tabs>
-<TabItem value="example_file" label="Full example of an action file" default>
+<TabItem value="complete_action" label="RGAction Completes the Action" default>
 
 ```cs
-public class RGPlayerMoveAction : RGAction
+public class Player : MonoBehavior
 {
-    
-    private Vector3? targetPosition;
-    private Rigidbody rigidbody;
-    public float speed = 5000f;
-    public float range = 1f;
+    public float jumpPower = 10f;
+    private RigidBody _rigidbody;
 
-    public void Awake()
+    void Start()
     {
-        rigidbody = GetComponent<Rigidbody>();
+        _rigidbody = GetComponent<Rigidbody>();
     }
 
-    public override string GetActionName()
+    [RGAction]
+    public void Jump()
     {
-        return "MoveToPosition";
+        _rigidbody.AddForce(Vector3.up * jumpPower, ForceMode.Impulse);
     }
+}
+```
+
+</TabItem>
+<TabItem value="action_with_update" label="RGAction Starts the Action">
+
+```cs
+public class Player : MonoBehavior
+{
+    public float speed = 100f;
+    public float range = 1f;
+    private RigidBody _rigidbody;
+    private Vector3? _targetPosition;
 
-    public override void StartAction(Dictionary<string, object> input)
+    void Start()
     {
-        var targetX = float.Parse(input["x"].ToString());
-        var targetY = float.Parse(input["y"].ToString());
-        var targetZ = float.Parse(input["z"].ToString());
-        targetPosition = new Vector3(targetX, targetY, targetZ);
+        _rigidbody = GetComponent<Rigidbody>();
     }
 
     public void Update()
     {
-        
         // If we are in range, reset the action
         if (targetPosition != null && Vector3.Distance((Vector3) targetPosition, transform.position) < range)
         {
             targetPosition = null;
         }
-        
+
         // Set the target velocity
         if (targetPosition != null)
         {
-            rigidbody.velocity = ((Vector3) targetPosition - transform.position).normalized * speed * Time.deltaTime;
+            Vector3 direction = ((Vector3)targetPosition - transform.position).normalized;
+            direction.y = 0;
+            float force = speed * Time.deltaTime;
+            _rigidbody.AddForce(direction * force);
         }
     }
+
+    [RGAction]
+    public void MoveToPosition(float x, float y, float z)
+    {
+        targetPosition = new Vector3(x, y, z);
+    }
 }
 ```
+
 </TabItem>
-</Tabs>
\ No newline at end of file
+</Tabs>
+
+:::info
+We recommend using primitive data types for `RGAction` parameters.
```

Comment:
I don't see the question anymore, but I remember seeing something about array support. I tried it out and it does not work. The Serializer class generates
```
return JsonUtility.FromJson<int[]>(paramJson);
```
which gives a compiler error. We can put that as a new task 👍